src/coreclr/nativeaot/Runtime/amd64/ExceptionHandling.asm

```diff
@@ -534,20 +535,25 @@ endif
         ;; if have shadow stack, then we need to reconcile it with the rsp change we have just made
         rdsspq  r9
         test    r9, r9
-        jz      NoSSP
-
-        ;; Find the shadow stack pointer for the frame we are going to restore to.
-        ;; The SSP we search is pointing to the return address of the frame represented
-        ;; by the passed in context. So we search for the instruction pointer from 
-        ;; the context and return one slot up from there.
-        ;; (Same logic as in GetSSPForFrameOnCurrentStack)
-        xor     r11, r11
-   @@:  inc     r11
-        cmp     [r9 + r11 * 8 - 8], r10
-        jne     @b
-
-        incsspq r11
-NoSSP:  jmp     rax
+        jz      SSP_OK
+
+        ;; assume common case
+        ;; in addition to the actual pops from the throwing site to the catching site,
+        ;; pop off DispatchEx, RhThrowEx, RhpThrowEx2 and the actual method that threw
+        add     r10, 4
```

Comment:
Instead of carefully figuring out how many levels to skip, it seems we could just record SSP in the RhThrowEx/RhpThrowEx.